pkg/driver/driver_gcp.go

```diff
@@ -84,16 +84,29 @@ func (d *GCPDriver) Create() (string, string, error) {
 
 	var disks = []*compute.AttachedDisk{}
 	for _, disk := range d.GCPMachineClass.Spec.Disks {
-		disks = append(disks, &compute.AttachedDisk{
-			AutoDelete: disk.AutoDelete,
-			Boot:       disk.Boot,
-			InitializeParams: &compute.AttachedDiskInitializeParams{
-				DiskSizeGb:  disk.SizeGb,
-				DiskType:    fmt.Sprintf("zones/%s/diskTypes/%s", zone, disk.Type),
-				Labels:      disk.Labels,
-				SourceImage: disk.Image,
-			},
-		})
+		var attachedDisk compute.AttachedDisk
+		if disk.Type == "SCRATCH" {
+			attachedDisk = compute.AttachedDisk{
+				AutoDelete: disk.AutoDelete,
+				Type:       disk.Type,
+				Interface:  disk.Interface,
+				InitializeParams: &compute.AttachedDiskInitializeParams{
+					DiskType: fmt.Sprintf("zones/%s/diskTypes/%s", zone, "local-ssd"),
+				},
+			}
+		} else {
+			attachedDisk = compute.AttachedDisk{
+				AutoDelete: disk.AutoDelete,
```

Comment:
same as above

Reply:
@tennix - Can we default this flag to true in the MCM?

Reply:
For persistent disks, is it better to default to false so if the VM is deleted there is still a chance to recover the data in persistent disk especially for production environment?

Reply:
> @tennix - Can we default this flag to true in the MCM?

Actually, at the moment, Gardener sets the `AutoDelete` to `true` for all the GCPMachineClasses, hence the current codepath should not impact us immediately.

>For persistent disks, is it better to default to false so if the VM is deleted there is still a chance to recover the data in persistent disk especially for production environment?

I would actually suggest to set the `AutoDelete=true` even for the persistent disks, and let the user change the behaviour if needed via MachineClass. 
- The change to the api type to `*bool` would be useful here as well.